### PR TITLE
feat(mobile): migrate BottomSheetFlashList to new lib

### PR DESCRIPTION
**Pull request by Nodonisko**
## Description

Use external lib which provides much more "native" experience. In future we should migrate also normal BottomSheet, should be pretty easy.

It also solved bug on Android where scrolling down in BottomSheet closes BottomSheet.

## Related Issue

Resolve 

## Screenshots:

https://github.com/user-attachments/assets/e043a2f1-fa65-45cb-bbbb-c8f5ae245274

### CODE REVIEW

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx [hunk 1]
```diff
@@ -39,107 +38,103 @@ const sheetContentContainerStyle = prepareNativeStyle<{
...
+    backgroundColor: utils.colors.borderDashed,
 }));
 
+const WindowOverlay = ({ children }: { children: ReactNode }) => {
+    if (Platform.OS === 'ios') {
+        return <FullWindowOverlay>{children}</FullWindowOverlay>;
+    }
+
+    return (
+        <View style={{ position: 'absolute', top: 0, left: 0, right: 0, bottom: 0 }}>
```
Comment (PeKne) [inline suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: Move the styles to separated styleSheet object please.

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx [hunk 2]
```diff
@@ -39,107 +38,103 @@ const sheetContentContainerStyle = prepareNativeStyle<{
...
 
-const sheetWithOverlayStyle = prepareNativeStyle(_ => ({
-    flex: 1,
-    justifyContent: 'flex-end',
+const handleStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.borderDashed,
 }));
 
+const WindowOverlay = ({ children }: { children: ReactNode }) => {
+    if (Platform.OS === 'ios') {
```
Comment (PeKne) [inline suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: Please add comment why is this platform specific handling needed.

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx [hunk 3]
```diff
@@ -39,107 +38,103 @@ const sheetContentContainerStyle = prepareNativeStyle<{
...
+    const bottomSheetModalRef = useRef<BottomSheetModal>(null);
+
+    const handleClose = useCallback(() => {
+        bottomSheetModalRef.current?.dismiss();
+        onClose(false);
+    }, [onClose]);
+
+    const insetBottom = Math.max(insets.bottom, DEFAULT_INSET_BOTTOM);
+
+    // ref
```
Comment (PeKne) [inline suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: Are these comments necessary?

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx [hunk 4]
```diff
@@ -39,107 +38,103 @@ const sheetContentContainerStyle = prepareNativeStyle<{
...
+                    {...props}
+                    onPress={handleClose}
+                    appearsOnIndex={0}
+                    disappearsOnIndex={-1}
+                />
+            )}
+            backgroundStyle={applyStyle(bottomSheetStyle)}
+            handleIndicatorStyle={applyStyle(handleStyle)}
+            // @ts-expect-error wrong type, doesn't expect children
+            containerComponent={({ children }) => <WindowOverlay>{children}</WindowOverlay>}
```
Comment (PeKne) [inline suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: Could you do this?
```suggestion
            containerComponent={WindowOverlay}

**Review by PeKne — APPROVED** [review]
Awesome, this library makes the bottom sheet implementation way lighter on our side 👏. It was tested on Android. Unfortunately, I am upgrading Xcode, so I can not test iOS now 🙃.

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx (hunk 1 above)
Comment (Nodonisko) [reply suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: fixed

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx (hunk 2 above)
Comment (Nodonisko) [reply suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: removed, it was not necessary

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx (hunk 3 above)
Comment (Nodonisko) [reply suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: removed

File: suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx (hunk 4 above)
Comment (Nodonisko) [reply suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx]: yes done

**Pull request merged**